ciphey/basemods/Checkers/quadgrams.py

```diff
@@ -0,0 +1,54 @@
+import logging
+import re
+from math import log10
+from typing import Dict, Optional
+
+from ciphey.iface import Checker, Config, ParamSpec, T, Translation, registry
+from rich.logging import RichHandler
+
+
+@registry.register
+class Quadgrams(Checker[str]):
+
+    """
+    Uses Quadgrams to determine plaintext
+    """
+
+    def check(self, ctext: T) -> Optional[str]:
+        logging.debug("Trying Quadgrams checker")
+        ctext = re.sub("[^A-Z]", "", ctext.upper())
+        quadgrams = self.QUADGRAMS_DICT
+        N = sum(quadgrams.values())
+        score = 0
+        for key in quadgrams.keys():
+            quadgrams[key] = float(quadgrams[key]) / N
+        floor = log10(0.01 / N)
+        for i in range(len(ctext) - 4 + 1):
+            if ctext[i : i + 4] in quadgrams:
+                score += quadgrams[ctext[i : i + 4]]
```

Comment:
Comment here explaining why?

Reply:
It takes quadgrams from `ctext` and then it checks if those quadgrams are in the list. If they are it adds the score of the quadgrams to the total score. I'll make a comment.